kb/how-to/rewrite-branch-for-coding-style.md

```diff
@@ -0,0 +1,163 @@
+
+1. First identify the new commits, and attach them to the commit preceding the code style switch. This is an ordinary Git rebase (`git rebase code-style-switch-commit~ mywork`).
+2. Iterate over the commits from the `mywork` branch after it forks from the target branch. For each commit, construct a new commit whose content is the restyled content of the old commit, and form a chain starting at the code style switch commit.
+
+Using diagrams in the same form as the Git manual, the original situation is:
+
+```
+      A---B---C mywork
+     /
+D---E---F---G---H target
+            ^
+            |
+            "Switch to the new code style"
+```
+
+The first phase attaches `mywork` just before the code style switch commit G:
+
+```
+          A'---B'---C' mywork
+         /
+D---E---F---G---H target
+            ^
+            |
+            "Switch to the new code style"
+```
+
+In the second phase, the migration successively constructs a new branch with restyled commits. First the migration restyles the content of A' to obtain A'':
+
+```
+            A'---B'---C' mywork
+           /
+          /   A'' new/mywork
+         /   /
+D---E---F---G---H target
+            ^
+            |
+            "Switch to the new code style"
+```
+
+The process continues with B' and so on:
+
+```
+            A'---B'---C' mywork
+           /
+          /   A''---B'' new/mywork
+         /   /
+D---E---F---G---H target
+            ^
+            |
+            "Switch to the new code style"
+```
+
+Once the migration reaches the end of the branch, it renames the new branch to the original name:
+
+```
+            A'---B'---C' old/mywork
+           /
+          /   A''---B''---C'' mywork
+         /   /
+D---E---F---G---H target
+            ^
+            |
+            "Switch to the new code style"
+```
+
+The branch `mywork` can now be pushed to a pull request.
+
+## When to migrate
+
+If you have not yet made a pull request, you can migrate your work at any time.
+
+If you have opened a pull request but review has not yet started on it, please migrate your branch to the new code style before review starts.
+
+If there is an ongoing review of a pull request, please communicate with the reviewer(s) to decide on a good time. Rebasing a branch during review is disruptive, because comments end up detached from the code they're about, so it's generally best not to do it. Typically the best time to migrate the branch is after the reviewer(s) have approved.
+
+## How to migrate
+
+### Requirements
+
+Please have the following tools and files available:
+
+* Git 2.17.0 or newer.
+* Uncrustify 0.75.1. Please note that older or newer versions are likely not to give the desired result.
```

Comment:
Note: should we have install instructions for Uncrustify 0.75.1 somewhere? Could be as short as "Run `git clone --branch uncrustify-0.75.1 https://github.com/uncrustify/uncrustify.git` then follow [the readme](https://github.com/uncrustify/uncrustify#build)."

Reply:
Calling for volunteers to write Windows and macOS instructions...